rj_gameplay/rj_gameplay/play/basic122.py

```diff
@@ -1,116 +1,138 @@
-
-        # Get list of all skills with assigned roles from tactics
-
-        skills = self.striker_tactic.tick(
-            world_state, role_results[self.striker_tactic]
-        )
-        skills += self.seek_left.tick(world_state, role_results[self.seek_left])
-        skills += self.seek_right.tick(world_state, role_results[self.seek_right])
-        skills += self.goalie_tactic.tick(world_state, role_results[self.goalie_tactic])
-        skills += self.wall_tactic_1.tick(world_state, role_results[self.wall_tactic_1])
-        skills += self.wall_tactic_2.tick(world_state, role_results[self.wall_tactic_2])
-
-        skill_dict = {}
-        skill_dict.update(role_results[self.striker_tactic])
-        skill_dict.update(role_results[self.seek_left])
-        skill_dict.update(role_results[self.seek_right])
-        skill_dict.update(role_results[self.goalie_tactic])
-        skill_dict.update(role_results[self.wall_tactic_1])
-        skill_dict.update(role_results[self.wall_tactic_2])
-        return skill_dict, skills
-
-    def is_done(self, world_state):
-        return self.striker_tactic.is_done(world_state)
+        world_state: stp.rc.WorldState,
+    ) -> List[RobotIntent]:
+        """
+        INIT: assign goalie, 2 seekers
+        WAIT_TO_PASS: loop until seekers reach pt
+        INIT_PASS: assign goalie, pass tac
+        PASSING: pass to one of two seekers, when pass done, shoot
+        PASSING_ASSIGN_ROLES: when pass tactic needs role assignment while psasing, assign roles for it, then go back to PASSING
+        INIT_SHOOT: assign goalie, striker, seeker
+        SHOOTING: striker shoots
+        DONE: shot taken
+        """
+
+        if self._state == State.INIT:
+            self.prioritized_tactics = [
+                goalie_tactic.GoalieTactic(world_state, 0),
+                basic_seek.BasicSeek(self._seek_pts[0], world_state),
+                basic_seek.BasicSeek(self._seek_pts[1], world_state),
+            ]
+
+            self.assign_roles(world_state)
+
+            self._state = State.WAIT_TO_PASS
+            return self.get_robot_intents(world_state)
+
+        elif self._state == State.WAIT_TO_PASS:
+
+            seek_1 = self.prioritized_tactics[1]
+            seek_2 = self.prioritized_tactics[2]
+
+            # TODO: is one tick delay issue?
+            if seek_1.is_done(world_state) and seek_2.is_done(world_state):
+                self._state = State.INIT_PASS
+
+            return self.get_robot_intents(world_state)
+
+        elif self._state == State.INIT_PASS:
+            init_passer_cost = stp.role.cost.PickClosestToPoint(world_state.ball.pos)
+            init_receiver_cost = stp.role.cost.PickClosestToPoint(self._seek_pts[0])
+            self.prioritized_tactics = [
+                goalie_tactic.GoalieTactic(world_state, 0),
+                pass_tactic.PassTactic(
+                    world_state, init_passer_cost, init_receiver_cost
```

Comment:
Is there a reason that we don't have seek tactic here?

Reply:
no, will add TODO